src/DataObjectAnnotator.php

```diff
@@ -61,11 +64,26 @@ class DataObjectAnnotator
     /**
      * DataObjectAnnotator constructor.
      * @throws NotFoundExceptionInterface
+     * @throws ReflectionException
      */
     public function __construct()
     {
         // Don't instantiate anything if annotations are not enabled.
         if (static::config()->get('enabled') === true && Director::isDev()) {
+            $extendableClasses = Config::inst()->getAll();
+            // We need to check all config to see if the class is extensible
+            // @todo find a cleaner method, this is already better than previous implementations though
+            foreach ($extendableClasses as $key => $configClass) {
+                if (isset($configClass['extensions']) && count($configClass['extensions']) > 0) {
+                    self::$extension_classes[] = ClassInfo::class_name($key);
+                }
+            }
+
+            // Because the tests re-instantiate the class every time
+            // We need to make it a unique array
+            // Also, it's not a bad practice, making sure the array is unique
+            self::$extension_classes = array_unique(self::$extension_classes);
```

Comment:
Although valid point, moving it to the setter is not ideal, because it would call the unique method for _every_ class that is  attempted to add...

What about a in_array() check? It would result in the same outcome, but the addition not being done?

Reply:
yup, that's fine too